### ISSUE TITLE
Rule `no-unexpected-multiline` does not detect case which line starts with `/`

### ISSUE THREAD

**Issue by lazex** [eslint/eslint#8469]
**Tell us about your environment**

* **ESLint Version:**
v3.19.0
* **Node Version:**
v6.10.1
* **npm Version:**
3.10.10
**What parser (default, Babel-ESLint, etc.) are you using?**
default
**Please show your full configuration:**

```
{
    "env": {
        "browser": true,
        "es6": true
    },
    "extends": "eslint:recommended",
    "rules": {
        "indent": [
            "error",
            "tab"
        ],
        "linebreak-style": [
            "error",
            "unix"
        ],
        "quotes": [
            "error",
            "double"
        ],
        "semi": [
            "error",
            "never"
        ],
        "no-console": [
            "off"
        ]
    }
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
var g = 1

[g].forEach(_ => _)

g = 2

`g`

var i = 3

/g/i.test("G") && console.log(g)
```

**What did you expect to happen?**
Three errors are detected as `no-unexpected-multiline`

- Before `g = 2`
- Before ``` `g` ```
- Before `/g/i.test("G") && console.log(g)`

**What actually happened? Please include the actual, raw output from ESLint.**
Only two errors were detected.

```
  3:1  error  Unexpected newline between object and [ of property access    no-unexpected-multiline
  5:5  error  Unexpected newline between template tag and template literal  no-unexpected-multiline

✖ 2 problems (2 errors, 0 warnings)
```

**Comment by not-an-aardvark** [comment]
This looks like a bug to me. Thanks for reporting.

**Comment by not-an-aardvark** [comment]
I'm not sure about the best way to detect this pattern. For the most part, I think this is only a problem when the code that looks like a regular expression has flags (since a syntax error will occur if it doesn't have flags). There are a few esoteric cases where that's not true:

```js
// This is a division operation
var x = a
/maybeRegex/
c;

// Maybe the user intended the middle line to be a bare regex literal?
// Seems like this expectation would be very uncommon
```

There are a few options that I can think of:

1. Disallow all division where the division operator is on a new line
  
    This could cause some false positives:
  
    ```js
    // this should probably be valid
    var fraction = numerator
      / denominator;
    ```

1. Disallow division where the division operator is on a new line, and the parent is also a division operator on the same line. This could also cause a few false positives:

    ```js
    var fractionOver2 = numerator
      / denominator / 2;
    ```

1. Disallow division where the division operator is on a new line, and the parent is also a division operator on the same line, and the right-hand side of the parent only contains valid flags, and there is no space between the second division operator and the "flags". In theory this could cause false negatives if the user was trying to create a regex and made a typo in the flags, but this is probably a very rare case. It could also cause false negatives if the user was trying to create a regex but accidentally put space before the flags, but this is also probably quite rare.

So right now I'm leaning toward option 3.
